clustered/server/src/main/java/org/ehcache/clustered/server/state/MessageTracker.java

```diff
@@ -80,6 +85,17 @@ public boolean isEmpty() {
    * Remove the contiguous seen msgIds from the nonContiguousMsgIds and update highestContiguousMsgId
    */
   private void reconcile() {
+
+    // This happens when a passive is started after Active has moved on and
+    // passive starts to see msgIDs starting from a number > 0.
+    if (highestContiguousMsgId == -1L && nonContiguousMsgIds.size() > 100) {
```

Comment:
Can't we add a field to the end of the message?  3.3.0 passives would not read it and use their existing logic, and would just skip over it... 3.3.1 passives would look for it but default to the 'magic' number if they don't see it.  So at least the problem would be fixed for fully 3.3.1 systems.